playground/bicep/BicepSample.AppHost/Program.cs

```diff
@@ -37,8 +37,8 @@
 
 var administratorLogin = builder.AddParameter("administratorLogin");
 var administratorLoginPassword = builder.AddParameter("administratorLoginPassword", secret: true);
-var pg = builder.AddPostgres("postgres2", administratorLogin, administratorLoginPassword)
-                .AsAzurePostgresFlexibleServer()
+var pg = builder.AddAzurePostgresFlexibleServer("postgres2")
+                .WithPasswordAuth(administratorLogin, administratorLoginPassword)
```

Comment:
Should this be `WithPasswordAuthentication`?

Reply:
The reason I chose these names (WithPasswordAuth here and WithAccessKeyAuth in Reds) is because they match the APIs from Azure.Provisioning (which I assume just match the names in the rest API).

https://github.com/dotnet/aspire/blob/e3e903ab9f648e7780abb1a5bd351bb7d393b6c2/src/Aspire.Hosting.Azure.PostgreSQL/AzurePostgresExtensions.cs#L192-L196

But looking again at Redis, the name is `DisableAccessKeyAuthentication`, which is inconsistent with the above Postgres APIs. So it is probably better we just completely follow .NET Design Guidelines with our APIs here and use full words.

Reply:
FYI - @tg-msft. I'm assuming this isn't something we can change/fix in Azure.Provisioning since it is just generated from the service APIs / typespec.

Reply:
We should do an API review, `WithAccessKey` is not intuitive. We should take the liberty here to add some opinions to this layer of the stack. We can do this in a follow up PR.

Reply:
As far as redis is concerned its a password.

Reply:
As far as Azure is concerned, it is an "access key":

![image](https://github.com/user-attachments/assets/47be5890-4f01-4192-bafc-fd05e084a02b)

Reply:
Specified as the password in the redis connection string.

Reply:
@philon-msft - is there a reason that the portal / ARM use the term "access key"?

Reply:
Azure Redis doesn't let you choose your own (like a "password") - instead it generates one for you (more like an "access key"). 
So in general Redis terminology it's a password, but for Azure Redis it's an access key.

Reply:
Correct, it's [`passwordAuth` on the wire](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2022-12-01/FlexibleServers.json#L744) and we're not looking to deviate from our automated translation of [the existing ARM experience](https://learn.microsoft.com/en-us/dotnet/api/azure.resourcemanager.postgresql.flexibleservers.models.postgresqlflexibleserverauthconfig.passwordauth?view=azure-dotnet#azure-resourcemanager-postgresql-flexibleservers-models-postgresqlflexibleserverauthconfig-passwordauth).  There's no way Azure.Provisioning would scale to all of Azure otherwise.

Reply:
> Azure Redis doesn't let you choose your own (like a "password") - instead it generates one for you (more like an "access key"). So in general Redis terminology it's a password, but for Azure Redis it's an access key.

Does Azure Redis let me add users with their own passwords and ACLs?

Reply:
Not standard Redis users+passwords, but you can add *Entra* users/apps/ManagedIdentities and assign them to roles for ACL functionality. See [doc](https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-azure-active-directory-for-authentication) for the Entra experience